### PR TITLE
Fixed Trip deletion to remove a trips subcollection when its being deleted

### PR DESCRIPTION
**Pull request by tosigame**
This PR updates the deleteTrip method to asynchronously delete all associated subcollections (stops, suggestions, announcements, expenses, users adn ect...) along with the trip itself. Each deletion task is launched within a coroutineScope to ensure that all related components are deleted before the trip document is removed from Firestore. This approach enhances the robustness of the deletion process and ensures a cleaner database state post-deletion. Additional logging has been added for better traceability and error handling is incorporated to manage potential exceptions effectively.

### CODE REVIEW

File: app/src/test/java/com/github/se/wanderpals/repository/TripsRepositoryTest.kt [hunk 1]
```diff
@@ -68,7 +136,7 @@ class TripsRepositoryTest {
 
         withTimeout(10000) {
           assertFalse(repository.addTripId(nonExistentTripId))
-          assertFalse(repository.removeTripId(nonExistentTripId))
+          assertTrue(repository.removeTripId(nonExistentTripId))
```
Comment (mtv47) [inline app/src/test/java/com/github/se/wanderpals/repository/TripsRepositoryTest.kt:139]: Why did this change to return true now ?

File: app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt [hunk 2]
```diff
@@ -1353,8 +1355,35 @@ open class TripsRepository(
       withContext(dispatcher) {
         try {
           Log.d("TripsRepository", "deleteTrip: Deleting trip")
-          removeTripId(tripId) // remove the trip from the user
-          tripsCollection.document(tripId).delete().await() // delete a given trip by its tripId
+
```
Comment (mtv47) [inline app/src/main/java/com/github/se/wanderpals/model/repository/TripsRepository.kt:1358]: Great use of multiple launch calls, didn't know it was possible. Nice that it's concurrent.

**Review by mtv47 — APPROVED** [review]
Small question but all good otherwise

File: app/src/test/java/com/github/se/wanderpals/repository/TripsRepositoryTest.kt (hunk 1 above)
Comment (tosigame) [reply app/src/test/java/com/github/se/wanderpals/repository/TripsRepositoryTest.kt:139]: The previous version of removeTripid, failed if it couldn't find the TripId,which we wished to delete, however this is not the case in mutliple other methods, which for consistency will return true, if the target of deletion doesn't exist. So I just fixed this method to follow the others. Hope this clears things out)

Comment (mtv47) [reply app/src/test/java/com/github/se/wanderpals/repository/TripsRepositoryTest.kt:139]: Thx for the very clear explanation

**Pull request merged**